compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -614,6 +614,33 @@ class Definitions {
       JavaSerializableClass.typeRef
     else
       ctx.requiredClassRef("scala.Serializable")
+
+   @threadUnsafe lazy val JavaEnumClass: ClassSymbol = {
+    val cls = ctx.requiredClass("java.lang.Enum")
```

Comment:
ping @sjrd, this might interact with Scala.js reimplementation of Enum: https://github.com/scala-js/scala-js/blob/master/javalanglib/src/main/scala/java/lang/Enum.scala

Reply:
Also it'd be nice to add a comment explaining why this hijacking is necessary.

Reply:
Hum, it's unclear how that would interact with the Scala.js implementation, indeed. However, that's not an immediate problem, because for now Dotty.js reuses the binaries of `java.lang.Enum.sjsir` from Scala.js for scalac.

Worse come to worst, we would hard-code the .sjsir of `java.lang.Enum` in the distant future where we can actually compile Scala.js' standard library with Dotty.js.